testing/integration/agent_long_running_leak_test.go

```diff
@@ -0,0 +1,296 @@
+	require.NoError(runner.T(), err, "got out: %s", string(out))
+
+	cmd = exec.Command("flog", "-t", "log", "-f", "apache_error", "-o", "/var/log/httpd/error_log", "-b", "50485760", "-p", "1048576")
+	out, err = cmd.CombinedOutput()
+	require.NoError(runner.T(), err, "got out: %s", string(out))
+
+	policyUUID := uuid.New().String()
+	unpr := false
+	installOpts := atesting.InstallOpts{
+		NonInteractive: true,
+		Force:          true,
+		Unprivileged:   &unpr,
+	}
+
+	fixture, err := define.NewFixture(runner.T(), define.Version())
+	require.NoError(runner.T(), err)
+	runner.agentFixture = fixture
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancel()
+
+	basePolicy := kibana.AgentPolicy{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+
+	// install system package
+	runner.InstallPackage(ctx, "system", "1.53.1", "agent_long_test_base_system_integ.json", uuid.New().String(), policyResp.ID)
+
+	// install cef
+	runner.InstallPackage(ctx, "apache", "1.17.0", "agent_long_test_apache.json", uuid.New().String(), policyResp.ID)
+
+}
+
+func (runner *ExtendedRunner) InstallPackage(ctx context.Context, name string, version string, cfgFile string, policyUUID string, policyID string) {
+	installPackage := kibana.PackagePolicyRequest{}
+
+	jsonRaw, err := os.ReadFile(cfgFile)
+	require.NoError(runner.T(), err)
+
+	err = json.Unmarshal(jsonRaw, &installPackage)
+	require.NoError(runner.T(), err)
+
+	installPackage.Package.Version = version
+	installPackage.ID = policyUUID
+	installPackage.PolicyID = policyID
+	installPackage.Namespace = "default"
+	installPackage.Name = fmt.Sprintf("%s-long-test-%s", name, policyUUID)
+	installPackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing %s package....", name)
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, installPackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+}
+
+func (runner *ExtendedRunner) TestHandleLeak() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "15m"
+	}
+
+	regex := regexp.MustCompile(`[\d]+`)
+	status, err := runner.agentFixture.ExecStatus(ctx)
+	require.NoError(runner.T(), err)
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(ctx)
+
+		cefMatch := "logfile-apache"
+		foundCef := false
```

Comment:
```suggestion
		apacheMatch := "logfile-apache"
		foundApache := false
```

The variable names are out of date.